packages/esm-service-queues-app/src/transition-latest-queue-entry/transition-latest-queue-entry.component.tsx

```diff
@@ -1,32 +1,36 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
 import { useLatestQueueEntry } from './transition-latest-queue-entry.resource';
+import AddPatientToQueueModal from './add-patient-to-queue-modal/add-patient-to-queue-entry.modal';
 import TransitionQueueEntryModal from '../queue-table/queue-entry-actions/transition-queue-entry.modal';
+import { type Visit } from '@openmrs/esm-framework';
 
 interface TransitionLatestQueueEntryProps {
-  patientUuid: string;
+  activeVisit: Visit;
   closeModal: () => void;
-  modalTitle?: string;
 }
 
-const TransitionLatestQueueEntry: React.FC<TransitionLatestQueueEntryProps> = ({
-  closeModal,
-  patientUuid,
-  modalTitle,
-}) => {
+const TransitionLatestQueueEntry: React.FC<TransitionLatestQueueEntryProps> = ({ closeModal, activeVisit }) => {
+  const patientUuid = activeVisit?.patient?.uuid;
   const { t } = useTranslation();
-  const { data: queueEntry, error, isLoading } = useLatestQueueEntry(patientUuid);
-
-  if (error || !queueEntry) {
-    return null;
-  }
+  const { data: queueEntry } = useLatestQueueEntry(patientUuid);
 
   return (
-    <TransitionQueueEntryModal
-      queueEntry={queueEntry}
-      closeModal={closeModal}
-      modalTitle={t('TransitionLatestQueueEntry', "Transition patient's latest queue")}
-    />
+    <>
+      {queueEntry ? (
+        <TransitionQueueEntryModal
```

Comment:
Modals are generally registered in the `routes.json` and triggered using the `showModal` function, see - https://o3-docs.openmrs.org/docs/modal-system

Reply:
Please make the change to make the modal be triggered using the modal system as commented previously. Then you won't have to render the modals this way.